### PR TITLE
[DataGrid] Ignore event from portal in cells

### LINKED ISSUE #1295: [DataGrid] Loose focus when opening a textbox portal inside a cell
When we display a modal from a button included in a datagrid, we can not keep the focus inside on a TextField, it remains on the datagrid behind.
We can also delete data in row of the datagrid

## Current Behavior 😯

When you try to write in a TextField in a modal display from datagrid, you lose focus directly and you can't continue, unless you spam. If you try to delete nothing happens, or rather something happens on the datagrid behind.

## Steps to Reproduce 🕹
bug_material_ui

Bug on the modal in the datagrid, no out with the button (same component)
Steps:

1. Install 4.0.0-alpha.23
2. try Modal in datagrid

## Your Environment 🌎

  System:
    OS: Windows 10 10.0.19041

  Binaries:
    Node: 12.18.0 - C:\Program Files\nodejs\node.EXE
    Yarn: Not Found
    npm: 7.6.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 89.0.4389.90
    Edge: Spartan (44.19041.423.0), Chromium (89.0.774.57)   
 npmPackages:
    @material-ui/core: ^4.11.3 => 4.11.3
    @material-ui/data-grid: ^4.0.0-alpha.23 => 4.0.0-alpha.23
    @material-ui/icons: ^4.11.2 => 4.11.2
    @material-ui/lab: ^4.0.0-alpha.57 => 4.0.0-alpha.57
    @material-ui/pickers: ^3.3.10 => 3.3.10
    @material-ui/styles:  4.11.3
    @material-ui/system:  4.11.3
    @material-ui/types:  5.1.0
    @material-ui/utils:  4.11.2
    @types/react:  17.0.3
    react: ^17.0.1 => 17.0.1
    react-dom: ^17.0.1 => 17.0.1

```
  Same on Brave, Firefox
```

### LINKED ISSUE #1305: [DataGrid] Events from cell's portal should be ignored
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

click leak modal

## Expected Behavior 🤔

Isolation.

## Steps to Reproduce 🕹

1. Open https://codesandbox.io/s/material-demo-forked-nps4b?file=/demo.tsx
2. Open a dialog
3. Click on it

I believe the regression comes from #1158

### PR DESCRIPTION
**Pull request by oliviertassinari**
Closes #1295.
Closes #1305.

This seems to cover all the issues. Assuming we will never add another source, maybe it's enough? Opened as a draft to run the CI and collect feedback.

### CODE REVIEW

**Comment by dtassone** [comment]
it should be ok to have the event bubble to the grid container.

**Comment by oliviertassinari** [comment]
I can add a regression test case if we agree with the approach. I didn't dive too much into the codebase, to try to find other events subscribed. What I did instead of click/key around to find any wrong behavior, couldn't spot any.

**Comment by oliviertassinari** [comment]
I should also link the React issue in the comment, for context

**Comment by oliviertassinari** [comment]
I will wait for @dtassone to come back before continue the effort in the issue.

**Comment by dtassone** [comment]
You should add a test for renderCell, and portals to make sure it works as expected but the approach looks correct

**Comment by oliviertassinari** [comment]
Ok, will do such

**Comment by oliviertassinari** [comment]
@dtassone done

File: packages/grid/data-grid/src/tests/rows.DataGrid.test.tsx [hunk 1]
```diff
@@ -57,4 +63,30 @@ describe('<DataGrid /> - Rows', () => {
       expect(getColumnValues()).to.deep.equal(['Mike-11', 'Jack-11', 'Mike-20']);
     });
   });
+
+  it('should ignore events coming from a portal in the cell', () => {
```
Comment (dtassone) [inline packages/grid/data-grid/src/tests/rows.DataGrid.test.tsx:67]: Not sure we have a test that shows events from a rendercell without portal

Comment (oliviertassinari) [reply packages/grid/data-grid/src/tests/rows.DataGrid.test.tsx:67]: I don't understand. What do you mean?

Comment (dtassone) [reply packages/grid/data-grid/src/tests/rows.DataGrid.test.tsx:67]: well you check that the event is blocked with portal. I don't think we have a test that check if the event go through without portal.

Comment (oliviertassinari) [reply packages/grid/data-grid/src/tests/rows.DataGrid.test.tsx:67]: Ok, I can add this

**Review by dtassone — APPROVED** [review]

**Comment by oliviertassinari** [comment]
The React's issue: https://github.com/facebook/react/issues/11387.

**Pull request merged**